opensearchapi/opensearchapi_response_internal_test.go

```diff
@@ -103,6 +105,50 @@ func TestAPIResponse(t *testing.T) {
 		}
 	})
 
+	t.Run("Error", func(t *testing.T) {
+		res = &Response{StatusCode: 201}
+
+		if err := res.Err(); err != nil {
+			t.Errorf("Unexpected error for response: %s", res.Status())
+		}
+
+		res = &Response{StatusCode: 403}
+
+		if err := res.Err(); err == nil {
+			t.Errorf("Expected error for response: %s", res.Status())
+		}
+
+		res = &Response{
+			StatusCode: 400,
+			Body: io.NopCloser(
+				strings.NewReader(`
+				{
+					"error":{
+						"root_cause":[{
+							"type":"resource_already_exists_exception",
+							"reason":"index [test/HU2mN_RMRXGcS38j3yV-VQ] already exists",
+							"index":"test",
+							"index_uuid":"HU2mN_RMRXGcS38j3yV-VQ"
+						}],
+						"type":"resource_already_exists_exception",
+						"reason":"index [test/HU2mN_RMRXGcS38j3yV-VQ] already exists",
+						"index":"test",
+						"index_uuid":"HU2mN_RMRXGcS38j3yV-VQ"
+					},
+					"status":400
+				}`)),
+		}
+
+		err := res.Err()
+		if err == nil {
+			t.Errorf("Expected error for response: %s", res.Status())
+		}
+		var errTest *Error
+		if !errors.As(err, &errTest) {
+			t.Errorf("Expected error to be of type opensearchapi.Error: %T", err)
+		}
```

Comment:
Ensure that `.Reason` and such are correct?

Reply:
Added an `if` to verify the content of the custom error.